ci/glci/model.py

```diff
@@ -401,14 +409,32 @@ class BuildCfg:
     alicloud_region: str
     alicloud_cfg_name: str
 
-
 @dataclasses.dataclass(frozen=True)
-class AzurePublishCfg:
+class AzureMarketplaceCfg:
     offer_id: str
     publisher_id: str
     plan_id: str
-    service_principal_cfg_name: str  # references secret in cicd cluster
-    storage_account_cfg_name: str
+
+
+@dataclasses.dataclass(frozen=True)
+class AzureServicePrincipalCfg:
+    tenant_id: str
+    client_id: str
+    client_secret: str
+
+
+@dataclasses.dataclass(frozen=True)
+class AzureStorageAccountCfg:
+    name: str
+    container_name: str
+    access_key: str
+
+
+@dataclasses.dataclass(frozen=True)
+class AzurePublishCfg:
+    marketplace: AzureMarketplaceCfg
+    service_principal: AzureServicePrincipalCfg
+    storate_account: AzureStorageAccountCfg
```

Comment:
typo; I suppose, this should rather read `storage_account`?

Reply:
done